### PR TITLE
[BI-77] create/ edit blog improvements

### PR DESCRIPTION
**Pull request by LAlex14**
- common component for the form of create blog and edit blog page 
- + the possibility of editing a cover image
- if there is no text in the content, the publish button is disabled

### CODE REVIEW

File: src/api/blogService.js [hunk 1]
```diff
@@ -69,8 +69,9 @@ export async function createBlog(data) {
 
 export async function updateBlog({ blogId, data }) {
   data.tags = JSON.stringify(createTagsArr(data.tags));
+  data = createFormData(data);
   try {
-    return await axios.put(`/api/restify/blogs/${blogId}`, data);
+    return await axios.post(`/api/restify/blogs/${blogId}`, data);
```
Comment (PascuMihai) [inline src/api/blogService.js:74]: is there a reason why this changed?

File: src/components/ReadingTime.vue [hunk 2]
```diff
@@ -18,8 +18,8 @@ export default {
   },
   computed: {
     time() {
+      if (!this.text) return 1;
       const words = this.text.split(/\s/g).length;
-      console.log(words);
       return Math.ceil(words / this.WORDS_PER_MINUTE);
```
Comment (PascuMihai) [inline src/components/ReadingTime.vue:23]: can use ternary operator `return this.text ? Math.ceil(words / this.WORDS_PER_MINUTE) : 1;`

File: src/components/UpdateBlog.vue [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+</template>
+
+<script>
+import TiptapEditor from '@/components/TiptapEditor/TiptapEditor';
+import get from 'lodash/get';
+
+export default {
+  name: 'UpdateBlog',
+  components: { TiptapEditor },
+  props: ['blog', 'loading'],
```
Comment (PascuMihai) [inline src/components/UpdateBlog.vue:77]: let's add the type for these props, it's safer that way

File: src/components/UpdateBlog.vue [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  data() {
+    return {
+      previewImage: this.blog.image,
+    };
+  },
+  computed: {
+    isContent() {
+      return this.blog.content.length - 7 > 0;
```
Comment (PascuMihai) [inline src/components/UpdateBlog.vue:85]: Is this logic because of the html tags?

**Review by PascuMihai — COMMENTED** [review]
here is no option from where i can erase my blog picture
image

File: src/api/blogService.js (hunk 1 above)
Comment (LAlex14) [reply src/api/blogService.js:74]: da, pentru ca nu putem face put cu form-data la edit pentru a edita cover image

File: src/components/UpdateBlog.vue (hunk 4 above)
Comment (LAlex14) [reply src/components/UpdateBlog.vue:85]: yes <p></p>

File: src/api/blogService.js [hunk 5]
```diff
@@ -68,9 +70,11 @@ export async function createBlog(data) {
 }
 
 export async function updateBlog({ blogId, data }) {
+  console.log(data);
```
Comment (Ceciliaa) [inline src/api/blogService.js:73]: Can we remove the console.log?

File: src/components/UpdateBlog.vue [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  },
+  data() {
+    return {
+      previewImage: this.blog.image,
+    };
+  },
+  computed: {
+    isContent() {
+      return this.blog.content.length - 7 > 0;
+    },
```
Comment (Ceciliaa) [inline src/components/UpdateBlog.vue:97]: What represent 7?
Save this value in a variable with a suggestive name

**Review by Ceciliaa — APPROVED** [review]

File: src/api/blogService.js [hunk 7]
```diff
@@ -7,6 +7,8 @@ function createFormData(data) {
   formData.append('tags', data.tags);
   formData.append('content', data.content);
   formData.append('image', data.image);
+  console.log(data);
```
Comment (PascuMihai) [inline src/api/blogService.js:10]: remove

File: src/components/UpdateBlog.vue [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+                  <span>{{ $t('general.button.chooseFile') }}</span>
+                </label>
+                <base-button
+                  v-if="blog.image && $route.name === 'CreateBlog'"
+                  @click="deleteCoverImg"
+                  size="sm"
+                  color="danger"
+                  outline
+                >
+                  Delete
```
Comment (PascuMihai) [inline src/components/UpdateBlog.vue:50]: translation

File: src/components/UpdateBlog.vue [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  name: 'UpdateBlog',
+  components: { TiptapEditor },
+  props: {
+    blog: Object,
+    loading: Boolean,
+  },
+  data() {
+    return {
+      previewImage: this.blog.image,
+      pTagLength: 7,
```
Comment (PascuMihai) [inline src/components/UpdateBlog.vue:92]: can this have a better name, contentCharCount or have a better name

File: src/components/UpdateBlog.vue [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  },
+};
+</script>
+
+<style scoped>
+#coverImgName {
+  width: 100px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (PascuMihai) [inline src/components/UpdateBlog.vue:119]: all of these classes can be from tailwind, that's why we installed it
w-24 whitespace-nowrap overflow-hidden ellipsis and is it's not in tailwind create it

**Review by PascuMihai — APPROVED** [review]

File: src/components/UpdateBlog.vue (hunk 10 above)
Comment (LAlex14) [reply src/components/UpdateBlog.vue:119]: nu mai e necesara stilizarea data, am sters-o de tot, pt ca nu mai exista acel element

**Pull request merged**